packages/cape/src/methods.ts

```diff
@@ -223,26 +232,51 @@ export abstract class Methods {
     }
   }
 
+  public async generateDataEncryptionKey(key?: string): Promise<DataEncryptionKey> {
+    if (key == null) {
+      key = await this.key();
+    }
+
+    const plaintext = forge.random.getBytesSync(32);
+
+    const ciphertext = await rsaEncrypt(plaintext, key);
+
+    return { plaintext, ciphertext: ciphertext };
+  }
+
   /**
-   * Encrypt the
-   * The returned key is stored in the `client` object
+   * Encrypt the input.
+   *
+   * Encrypt takes options which allow you customize what RSA key or
+   * data encryption key (dek) is used. You cannot pass a custom RSA key
+   * and dek. This function will return an error if you do.
    *
    * @example
    * ```ts
    * const myInput = <YOUR_INPUT>;
    * const client = new Cape({ authToken: 'my-auth-token' });
    * await key = client.key();
-   * const encrypted = client.encrypt(myInput)
+   * const encrypted = client.encrypt(myInput, { key })
    *
    * ```
    */
-  public async encrypt(input: string, key?: string): Promise<string> {
-    let k = key;
-    if (k == null) {
-      k = await this.key();
+  public async encrypt(input: string, options: EncryptOptions = { key: undefined, dek: undefined }): Promise<string> {
+    if (options.key != null && options.dek != null) {
+      throw Error('cannot pass key and dek to this function');
+    }
+
+    if (options.dek != null) {
+      // TODO set this.saveDataEncryptedkey or something...
+      throw Error('not implemented');
     }
 
-    return await capeEncrypt(k, input);
+    if (this.savedDataEncryptedKey == null) {
```

Comment:
It seems we are just storing one data key, but we talked about having a key/value based on the rsa public key. In other words, we would create and store one data key for each public key

Reply:
Couldn't remember what we decided there works for me. Thanks!